### PR TITLE
Add in ino for file objects.

### PR DESCRIPTION
**Pull request by gnikonorov**
Partial resolution for https://github.com/fakefs/fakefs/issues/321; add the `ino` method for `File::Stat`.

### CODE REVIEW

File: lib/fakefs/fake/file.rb [hunk 1]
```diff
@@ -6,13 +6,47 @@ class FakeFile
...
+      # please see: http://iacobson.net/beware-of-ruby-class-variables/
+      class << self
+        attr_accessor :freed_inodes
+        attr_accessor :next_inode_num
+      end
+
+      # This method should only be used for tests
+      # When called, it will reset the current inode information of the FakeFS
+      def self.clear_inode_info_for_tests
+        Inode.freed_inodes = []
```
Comment (grosser) [inline lib/fakefs/fake/file.rb]: prefer `self.` so a superclass would also work properly

Comment (grosser) [reply lib/fakefs/fake/file.rb]: can keep this in the `class << self` block

File: lib/fakefs/fake/file.rb [hunk 2]
```diff
@@ -6,13 +6,47 @@ class FakeFile
...
+        Inode.next_inode_num = 0
+      end
+
+      def assign_inode_num
+        new_inode = -1
+        if !Inode.freed_inodes.empty?
```
Comment (grosser) [inline lib/fakefs/fake/file.rb]: `self.class.`

Comment (grosser) [reply lib/fakefs/fake/file.rb]: flip the if/else instead of making negative assertions ... or use .any?

File: lib/fakefs/fake/file.rb [hunk 3]
```diff
@@ -6,13 +6,47 @@ class FakeFile
...
+          new_inode = Inode.freed_inodes.shift
```
Comment (grosser) [inline lib/fakefs/fake/file.rb]: also can make this atomic by doing `if inode = self.class.freed_inodes.shift`

**Comment by grosser** [comment]
looks about right :)

**Comment by gnikonorov** [comment]
@grosser - if this looks OK, I will fork this to continue implementing the remainder of https://github.com/fakefs/fakefs/issues/321.

**Comment by grosser** [comment]
looks good! ... is this ready to merge on it's own ?

**Comment by gnikonorov** [comment]
@grosser - this is ready to merge. After the merge `File::Stat(<FILE_NAME>).ino` will be exposed for files **only**.

**Comment by grosser** [comment]
I'll not do a release now since you said you got another PR planned

**Pull request merged**